app/services/ocrService.py

```
@@ -19,25 +20,100 @@
+    gcs_filename: Optional[str] = None,
 ) -> Dict[str, Any]:
     if not file_bytes:
         raise OcrError("빈 파일입니다.")
     if content_type not in ALLOWED_MIME:
         allowed = ", ".join(sorted(ALLOWED_MIME))
         raise OcrError(f"지원하지 않는 형식입니다. 허용: {allowed}")
 
+    # GCS에 저장
+    if save_to_gcs:
+        if not user_hash:
+            raise OcrError("save_to_gcs=True일 때 user_hash가 필요합니다.")
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)